### PR TITLE
Sklearn hp tuning

### PR DESCRIPTION
**Pull request by happyhuman**

### CODE REVIEW

File: .kokoro/tests/training/sklearn_structured_hp_tuning.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+set -eo pipefail
+
+
+download_files() {
```
Comment (gogasca) [inline .kokoro/tests/training/sklearn_structured_hp_tuning.sh]: This seems to be a function that we can maybe generalize for all tests and remove from here?

File: training/sklearn/structured/hp_tuning/setup.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from setuptools import find_packages
+from setuptools import setup
+
+REQUIRED_PACKAGES = [
+    'tensorflow==1.13.1',
```
Comment (gogasca) [inline training/sklearn/structured/hp_tuning/setup.py]: TF 1.14.0?

File: training/sklearn/structured/hp_tuning/setup.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    'scikit-learn>=0.20.2',
```
Comment (gogasca) [inline training/sklearn/structured/hp_tuning/setup.py]: 0.21.3 ?

File: training/sklearn/structured/hp_tuning/README.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
+# Training with scikit-learn - Hyperparameter Tuning
```
Comment (gogasca) [inline training/sklearn/structured/hp_tuning/README.md]: Can we add more details as to which parameters we tune and the results of each run ?

File: training/sklearn/structured/hp_tuning/README.md [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+If you were able to run [train-cloud.sh](./scripts/train-cloud.sh) successfully, you have
+created and stored some files in your GCS bucket. You may simply remove them by running
+
+```bash
+source ./scripts/cleanup.sh
+```
+
+## How Hyperparameter Tuning Works
+
+While you may use `GridSearchCV` to tune your hyperparameters, AI Platform offers a better solution.
```
Comment (gogasca) [inline training/sklearn/structured/hp_tuning/README.md]: Link to supported options in sklearn?

File: training/sklearn/structured/hp_tuning/setup.py (hunk 3 above)
Comment (happyhuman) [reply training/sklearn/structured/hp_tuning/setup.py]: Not yet supported by CMLE 1.14 which supports these:

TensorFlow 1.14.0
scikit-learn 0.20.2
XGBoost 0.81

File: training/sklearn/structured/hp_tuning/README.md (hunk 4 above)
Comment (happyhuman) [reply training/sklearn/structured/hp_tuning/README.md]: Added

File: training/sklearn/structured/hp_tuning/README.md (hunk 5 above)
Comment (happyhuman) [reply training/sklearn/structured/hp_tuning/README.md]: Good point. Made that change.

File: .kokoro/tests/training/sklearn_structured_hp_tuning.sh (hunk 1 above)
Comment (happyhuman) [reply .kokoro/tests/training/sklearn_structured_hp_tuning.sh]: Fair point. Perhaps we should do it after this PR i merged, and for all the samples. What do you think?

File: training/sklearn/structured/hp_tuning/setup.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from setuptools import find_packages
+from setuptools import setup
+
+REQUIRED_PACKAGES = [
+    'tensorflow==1.14.1',
```
Comment (gogasca) [inline training/sklearn/structured/hp_tuning/setup.py]: 1.14.0 is the correct one, otherwise test will fail

**Comment by happyhuman** [comment]
@gogasca I had to make one small change, so the merge requires your blessing one more time.  :)

**Pull request merged**